### PR TITLE
Make  `Numpy` args compitable with  `torch` function.

### LINKED ISSUE #17950: Adding support for `NumPy` arguments for  ` PyTorch`  at  torch_frontend
## Issue Description

Currently, `PyTorch's` backend includes default argument name translations to ensure compatibility with `NumPy`. However, the similar functionality is missing in `ivy's torch_frontend`. This issue proposes adding a decorator to the `ivy` library to provide compatibility between `torch` functions and `numpy` arguments.

### Background

The existing [code in PyTorch's backend](https://github.com/pytorch/pytorch/blob/223f23292873940d1028730fb381d6cf73aac404/torch/csrc/utils/python_arg_parser.cpp#L54) demonstrates default translations for certain arguments, such as "dim" to "axis", "keepdim" to "keepdims", "input" to "x", "a", or "x1", and "other" to "x2". This behavior ensures that `PyTorch` functions align with `NumPy` conventions and makes it easier for users familiar with `NumPy` to work with `PyTorch`.

### Proposed Solution

The proposed solution involves adding a decorator called `torch_to_numpy_style_args` to the `ivy` library. This decorator will allow the translation of argument names from `PyTorch` style to `NumPy` style. It will handle the mapping of argument names such as "dim", "keepdim", "input", and "other" to their corresponding `torch_frontend` names, keeping the behavior similar to how `PyTorch` does it. 

### Implementation Details

The `torch_to_numpy_style_args` decorator will be implemented as follows:

```python
# path: ivy/functional/frontends/torch/func_wrapper.py

def torch_to_numpy_style_args(func):
    """
    Decorator for translating argument names from PyTorch style to NumPy style.
    """
    numpy_compatibility_arg_names = {
        "dim": ["axis"],
        "keepdim": ["keepdims"],
        "input": ["x", "a", "x1"],
        "other": ["x2"],
    }

    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        # Translate PyTorch-style args to NumPy-style args
        new_kwargs = {}
        for k, v in kwargs.items():
            ivy_names = numpy_compatibility_arg_names.get(k, [k])
            for ivy_name in ivy_names:
                new_kwargs[ivy_name] = v
        # Translate argument names explicitly
        for pytorch_name, numpy_names in numpy_compatibility_arg_names.items():
            for numpy_name in numpy_names:
                if numpy_name in new_kwargs:
                    new_kwargs[pytorch_name] = new_kwargs.pop(numpy_name)
        return func(*args, **new_kwargs)

    return wrapper
```
### Files to be modified by this decorator :
- [x] https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/torch/reduction_ops.py
- [ ] https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/torch/indexing_slicing_joining_mutating_ops.py
- [x] https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/torch/tensor.py

### PR DESCRIPTION
**Pull request by akshatvishu**
Close #17950

Also, fixed `squeeze` method not getting properly called at `torch_frontend.median` implementation.

As discussed with @AnnaTz  ; to provide the same compatibility that `torch` offers within its framework to allow `Numpy` args as an alias for `torch` args, we want to add the same functionality to `ivy`.

Currently, this checks compatibility for all functions at  [ivy/functional/frontends/torch/reduction_ops.py](https://github.com/unifyai/ivy/blob/master/ivy/functional/frontends/torch/reduction_ops.py)

### CODE REVIEW

**Comment by AnnaTz** [comment]
I'd say you can go ahead and add the decorator to all torch frontend functions that have one of the aliased arguments.

File: ivy/functional/frontends/torch/func_wrapper.py [hunk 1]
```diff
@@ -130,3 +130,55 @@ def outputs_to_native_arrays_torch(*args, **kwargs):
...
+#         # Translate PyTorch-style args to NumPy-style args
+#         new_kwargs = {}
+#         for k, v in kwargs.items():
+#             ivy_names = numpy_compatibility_arg_names.get(k, [k])
+#             for ivy_name in ivy_names:
+#                 new_kwargs[ivy_name] = v
+#         if 'axis' in new_kwargs:
+#             new_kwargs['dim'] = new_kwargs.pop('axis')
+#         return func(*args, **new_kwargs)
+#     return wrapper
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/func_wrapper.py:184]: Also, I guess you forgot to remove the commented out part?

Comment (akshatvishu) [reply ivy/functional/frontends/torch/func_wrapper.py:184]: Sorry! It was part of an earlier testing code !

**Comment by AnnaTz** [comment]
One ore thing, ideally we would like to have a test for the decorator as well. You can have a look at the rest of the `func_wrapper` tests I have implemented if you'd like to do this yourself. Otherwise I can do this.

**Comment by akshatvishu** [comment]
> One ore thing, ideally we would like to have a test for the decorator as well. You can have a look at the rest of the `func_wrapper` tests I have implemented if you'd like to do this yourself. Otherwise I can do this.

The test for it should be added at `ivy_tests/test_ivy/test_misc/test_func_wrapper.py` ?

**Comment by AnnaTz** [comment]
> > One ore thing, ideally we would like to have a test for the decorator as well. You can have a look at the rest of the `func_wrapper` tests I have implemented if you'd like to do this yourself. Otherwise I can do this.
> 
> The test for it should be added at `ivy_tests/test_ivy/test_misc/test_func_wrapper.py` ?

`ivy_tests/test_ivy/test_frontends/test_torch/test_func_wrapper.py`

File: ivy/functional/frontends/torch/func_wrapper.py [hunk 2]
```diff
@@ -130,3 +130,30 @@ def outputs_to_native_arrays_torch(*args, **kwargs):
...
+            for ivy_name in ivy_names:
+                new_kwargs[ivy_name] = v
+        # Translate argument names explicitly
+        for pytorch_name, numpy_names in numpy_compatibility_arg_names.items():
+            for numpy_name in numpy_names:
+                if numpy_name in new_kwargs:
+                    new_kwargs[pytorch_name] = new_kwargs.pop(numpy_name)
+        return func(*args, **new_kwargs)
+
+    return wrapper
```
Comment (AnnaTz) [inline ivy/functional/frontends/torch/func_wrapper.py:159]: I would prefer this:
```
def numpy_to_torch_style_args(func):
    """Convert argument names from NumPy style to PyTorch style."""
    numpy_compatible_args = {
        "axis": "dim",
        "keepdims": "keepdim",
        "x": "input",
        "a": "input",
        "x1": "input",
        "x2": "other",
    }

    @functools.wraps(func)
    def wrapper(*args, **kwargs):
        new_kwargs = copy.deepcopy(kwargs)
        for key in kwargs.keys():
            if key in numpy_compatible_args:
                new_kwargs[numpy_compatible_args[key]] = kwargs[key]
                del new_kwargs[key]
        return func(*args, **new_kwargs)

    return wrapper
```
1. The way I see this we are converting numpy style arguments to torch style, not the other way around.
2. This way we only use 1 for-loop.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/torch/func_wrapper.py (hunk 2 above)
Comment (akshatvishu) [reply ivy/functional/frontends/torch/func_wrapper.py:156]: Thank you for your feedback! I've made the requested changes to the code. Also, slightly changed the logic form using `deepcopy`  and instead using `dictionary comprehension`.

**Review by AnnaTz — APPROVED** [review]

**Comment by akshatvishu** [comment]
Hey @AnnaTz , thanks  for the review and all the  suggestions! Had a blast working on this one. Going to do a double-check on the rest of the files at `torch_frontend` to make sure I didn't skip over anything. Catch you in the next PR!

**Pull request merged**